### PR TITLE
off-chain attestation fixes

### PR DESCRIPTION
**Pull request by oleggrib**
@nicktaras , @micwallace , can you check my updates.
I had to update it to work OK with devconnect

I used those updates with UMD version.

### CODE REVIEW

**Comment by nicktaras** [comment]
Hi Oleh,

Could I just check. Are these changes global, for all future use cases?

E.g. No more use of Magic Link?

Thanks

Sent from my iPhone

On 15 May 2022, at 6:58 pm, sonarcloud[bot] ***@***.***> wrote:

﻿

Kudos, SonarCloud Quality Gate passed!    [Quality Gate passed] <https://sonarcloud.io/dashboard?id=TokenScript_token-negotiator&pullRequest=204>

[Bug]<https://sonarcloud.io/project/issues?id=TokenScript_token-negotiator&pullRequest=204&resolved=false&types=BUG> [A] <https://sonarcloud.io/project/issues?id=TokenScript_token-negotiator&pullRequest=204&resolved=false&types=BUG>  0 Bugs<https://sonarcloud.io/project/issues?id=TokenScript_token-negotiator&pullRequest=204&resolved=false&types=BUG>
[Vulnerability]<https://sonarcloud.io/project/issues?id=TokenScript_token-negotiator&pullRequest=204&resolved=false&types=VULNERABILITY> [A] <https://sonarcloud.io/project/issues?id=TokenScript_token-negotiator&pullRequest=204&resolved=false&types=VULNERABILITY>  0 Vulnerabilities<https://sonarcloud.io/project/issues?id=TokenScript_token-negotiator&pullRequest=204&resolved=false&types=VULNERABILITY>
[Security Hotspot]<https://sonarcloud.io/project/security_hotspots?id=TokenScript_token-negotiator&pullRequest=204&resolved=false&types=SECURITY_HOTSPOT> [A] <https://sonarcloud.io/project/security_hotspots?id=TokenScript_token-negotiator&pullRequest=204&resolved=false&types=SECURITY_HOTSPOT>  0 Security Hotspots<https://sonarcloud.io/project/security_hotspots?id=TokenScript_token-negotiator&pullRequest=204&resolved=false&types=SECURITY_HOTSPOT>
[Code Smell]<https://sonarcloud.io/project/issues?id=TokenScript_token-negotiator&pullRequest=204&resolved=false&types=CODE_SMELL> [A] <https://sonarcloud.io/project/issues?id=TokenScript_token-negotiator&pullRequest=204&resolved=false&types=CODE_SMELL>  1 Code Smell<https://sonarcloud.io/project/issues?id=TokenScript_token-negotiator&pullRequest=204&resolved=false&types=CODE_SMELL>

[No Coverage information]<https://sonarcloud.io/component_measures?id=TokenScript_token-negotiator&pullRequest=204> No Coverage information
[0.0%]<https://sonarcloud.io/component_measures?id=TokenScript_token-negotiator&pullRequest=204&metric=new_duplicated_lines_density&view=list> 0.0% Duplication<https://sonarcloud.io/component_measures?id=TokenScript_token-negotiator&pullRequest=204&metric=new_duplicated_lines_density&view=list>

**Comment by oleggrib** [comment]
@nicktaras  yes, its permanent updates.
MagicLink make us more difficulties and less security.

File: src/outlet/auth-handler.ts [hunk 1]
```diff
@@ -152,8 +155,8 @@ export class AuthHandler {
 		}
 
 		if (
-			!event?.data?.attestation
-            || !event?.data?.requestSecret
+			!event.data.hasOwnProperty('attestation')
+            || !event.data.hasOwnProperty('requestSecret')
```
Comment (micwallace) [inline src/outlet/auth-handler.ts:159]: @oleggrib maybe we can just remove this check?

and replace the next lines with:
```
		this.attestationBlob = event.data?.attestation;
		this.attestationSecret = event.data?.requestSecret;
```

Comment (oleggrib) [reply src/outlet/auth-handler.ts:159]: we can use your code
```
!event?.data?.attestation
|| !event?.data?.requestSecret
```
and we need a check if those values are set, if not then its wrong postMessage

Comment (nicktaras) [reply src/outlet/auth-handler.ts:159]: Either looks good.

Comment (nicktaras) [reply src/outlet/auth-handler.ts]: I've applied your change @micwallace. Testing the full PR via Github pages examples and works well on Chrome and Safari.

**Pull request merged**